tests/mma/mmatest.f90

```diff
@@ -0,0 +1,205 @@
+program mmatest
+!===========================================================================!
+!       This program is used to test the MMA solver implementation          !
+! This solves two different test problems using the MMA implementation in   !
+! the file mma.f90. Note that lapack function DGESV is used to solve linear !
+! system in mma.f90. Also num_types.f90 is used for data type configuration.!
+! Run by: (gfortran mmatest.f90 num_types.f90 mma.f90 -llapack)             !
+! ------------------------------------------------------------------------- !
+!                                                                           !
+!        TEST_CASE_1:                                                       !
+!    Problem 1 in "https://people.kth.se/~krille/originalmma.pdf"           !
+!                                                                           !
+!                                                                           !
+!        TEST_CASE_2:                                                       !
+!    Problem 1 in "https://people.kth.se/~krille/originalmma.pdf"           !
+!                                                                           !
+!                                                                           !
+! ------------------------------------------------------------------------- !
+!                                                                           !
+! To switch between the above two test cases:                               !
+! uncomment/comment the lines 36-37 and 40-41.                              !
+! uncomment/comment the lines 59-66 and 76-83.                              !
+! uncomment/comment the lines 106-111 and 115-120.                          !
+! Also, in file mma.f90 in "subroutine mma_init(this, x, n, m)",            ! 
+! uncomment/comment lines 166-172 and 175-181.                              !
```

Comment:
Please never rely on commenting or uncommenting stuff, espetially in the
initializer of mma. If it need different parameters for different cases, then it
should have inputs to control that.